src/ResourceManager/Profile/Commands.Profile/Errors/AzureExceptionRecord.cs

```diff
@@ -0,0 +1,39 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Profile.Errors
+{
+    public class AzureExceptionRecord : AzureErrorRecord
+    {
+        public AzureExceptionRecord(Exception exception, ErrorRecord record, bool inner = false) : base(record)
+        {
+            Message = exception.Message;
+            HelpLink = exception.HelpLink;
+            StackTrace = exception.StackTrace;
+            Exception = exception;
+        }
+
+        public bool InnerException { get; set; }
```

Comment:
@markcowl nit: can we put an empty line between `InnerException` and `Exception`?